# DATASET diagnosis model M2


El p

In [18]:
import os
import sys
sys.path.append(os.path.abspath(".."))
from scripts.useful_functions import how_many_files, csv_shape, unify_images, remove_unlabeled_images, unify_folders, merge_csv_files

In [19]:
cantidad_class0_test = how_many_files(r"E:\Proyects Python based\ProyectoAvanzado2\data\raw\M1\Test Images_new\Test Images\class_0")
cantidad_class1_test = how_many_files(r"E:\Proyects Python based\ProyectoAvanzado2\data\raw\M1\Test Images_new\Test Images\class_1")
cantidad_class0_train = how_many_files(r"E:\Proyects Python based\ProyectoAvanzado2\data\raw\M1\Train Images_new\Train Images\class_0")
cantidad_class1_train = how_many_files(r"E:\Proyects Python based\ProyectoAvanzado2\data\raw\M1\Train Images_new\Train Images\class_1")
print("Cantidad de imágenes de test clase 0: ", cantidad_class0_test)
print("Cantidad de imágenes de test clase 1: ", cantidad_class1_test)
print("Cantidad de imágenes de train clase 0: ", cantidad_class0_train)
print("Cantidad de imágenes de train clase 1: ", cantidad_class1_train)

Cantidad de imágenes de test clase 0:  1805
Cantidad de imágenes de test clase 1:  1860
Cantidad de imágenes de train clase 0:  7220
Cantidad de imágenes de train clase 1:  7425


In [20]:
total_test_img = cantidad_class0_test + cantidad_class1_test
total_train_img = cantidad_class0_train + cantidad_class1_train
print("El total de imágenes de test es: ", total_test_img)
print("El total de imágenes de train es: ", total_train_img)

El total de imágenes de test es:  3665
El total de imágenes de train es:  14645


In [21]:
test_csv = r"E:\Proyects Python based\ProyectoAvanzado2\data\processed\data_test.csv"
train_csv = r"E:\Proyects Python based\ProyectoAvanzado2\data\processed\data_train.csv"

print("test has: ", csv_shape(test_csv))
print("train has: ", csv_shape(train_csv))
test_csv_rows = csv_shape(test_csv)[0]
train_csv_rows = csv_shape(train_csv)[0]

if test_csv_rows == total_test_img:
    print("El número de filas del CSV de test coincide con el número de imágenes de test.") 
if train_csv_rows == total_train_img:
    print("El número de filas del CSV de train coincide con el número de imágenes de train.") 
labeled_images = test_csv_rows + train_csv_rows
print("El número total de imágenes etiquetadas es: ", labeled_images)

test has:  (734, 2)
train has:  (2930, 2)
El número total de imágenes etiquetadas es:  3664


In [22]:
# El número real de etiquetas es el número de filas menos 1 por el encabezado.
test_csv_rows_real = test_csv_rows - 1
train_csv_rows_real = train_csv_rows - 1
print("Número real de imágenes etiquetadas de test:", test_csv_rows_real)
print("Número real de imágenes etiquetadas de train:", train_csv_rows_real)
print("Número total real de imágenes etiquetadas:", test_csv_rows_real + train_csv_rows_real)

Número real de imágenes etiquetadas de test: 733
Número real de imágenes etiquetadas de train: 2929
Número total real de imágenes etiquetadas: 3662


Now that we know hoy many images are actually labeled, we can create a M3 dataset for diagnosis with the actual images

Train: 70% → ~2563 imágenes

Val: 15% → ~549 imágenes

Test: 15% → ~550 imágenes

In [23]:
# Unificar class_0 y class_1 en dos carpetas M3: train y test.
# Replace the origen paths with the actual M3 source folders if they differ.
#Para primra carrpeta
class0_test = r"E:\Proyects Python based\ProyectoAvanzado2\data\raw\M1\Test Images_new\Test Images\class_0"
class1_test = r"E:\Proyects Python based\ProyectoAvanzado2\data\raw\M1\Test Images_new\Test Images\class_1"
#Para segunda carpeta
class0_train = r"E:\Proyects Python based\ProyectoAvanzado2\data\raw\M1\Train Images_new\Train Images\class_0"
class1_train = r"E:\Proyects Python based\ProyectoAvanzado2\data\raw\M1\Train Images_new\Train Images\class_1"


m3_train_destino = r"E:\Proyects Python based\ProyectoAvanzado2\data\raw\M3_diagnosis\train"
m3_test_destino = r"E:\Proyects Python based\ProyectoAvanzado2\data\raw\M3_diagnosis\test"


print("Unificando TRAIN...")
unify_images(class1_train, m3_train_destino)
unify_images(class0_train, m3_train_destino)
print("\nUnificando TEST...")
unify_images(class0_test, m3_test_destino)
unify_images(class1_test, m3_test_destino)

Unificando TRAIN...
Resumen:
✔️ Imágenes copiadas: 7348
⚠️ Duplicados por nombre eliminados: 0
⚠️ Duplicados por contenido eliminados: 77
Resumen:
✔️ Imágenes copiadas: 7219
⚠️ Duplicados por nombre eliminados: 0
⚠️ Duplicados por contenido eliminados: 1

Unificando TEST...
Resumen:
✔️ Imágenes copiadas: 1805
⚠️ Duplicados por nombre eliminados: 0
⚠️ Duplicados por contenido eliminados: 0
Resumen:
✔️ Imágenes copiadas: 1854
⚠️ Duplicados por nombre eliminados: 0
⚠️ Duplicados por contenido eliminados: 6


In [34]:

cantidad_final_train = how_many_files(m3_train_destino)
cantidad_final_test = how_many_files(m3_test_destino)
print("Cantidad de imágenes en M3 train: ", cantidad_final_train)
print("Cantidad de imágenes en M3 test: ", cantidad_final_test)

Cantidad de imágenes en M3 train:  2851
Cantidad de imágenes en M3 test:  727


In [31]:
remove_unlabeled_images(train_csv, m3_train_destino)
remove_unlabeled_images(test_csv, m3_test_destino)

Resumen limpieza: 2851 imágenes revisadas en E:\Proyects Python based\ProyectoAvanzado2\data\raw\M3_diagnosis\train
✔️ Conservadas: 2851
🗑️ Eliminadas: 0
Resumen limpieza: 727 imágenes revisadas en E:\Proyects Python based\ProyectoAvanzado2\data\raw\M3_diagnosis\test
✔️ Conservadas: 727
🗑️ Eliminadas: 0


(0, 727, 727)

In [26]:
total_final_imagenes_etiquetadas = how_many_files(m3_train_destino) + how_many_files(m3_test_destino)
print("Número total de imágenes etiquetadas en M3: ", total_final_imagenes_etiquetadas)

Número total de imágenes etiquetadas en M3:  3578


We loss aproximate (~2.3%) of the labeled images, so we are going to keep going with the  3578, since the dataset has a solid number to be trained.

In [36]:
print("=== Unificando M3_diagnosis ===\n")

# Rutas base
m3_base = r"E:\Proyects Python based\ProyectoAvanzado2\data\processed\M3_diagnosis"
m3_imagenes_destino = os.path.join(m3_base, "imagenes")

# Limpiar carpeta si existe para evitar duplicados
if os.path.exists(m3_imagenes_destino):
    import shutil
    print("Limpiando carpeta de imágenes existente...")
    shutil.rmtree(m3_imagenes_destino)

# Crear carpeta limpia
os.makedirs(m3_imagenes_destino, exist_ok=True)

print("1. Unificando carpetas de imágenes...")
unify_folders(m3_train_destino, m3_test_destino, m3_imagenes_destino)

# Crear CSV unificado
m3_csv_unificado = os.path.join(m3_base, "data_unificado.csv")
print("\n2. Combinando archivos CSV...")
merge_csv_files(train_csv, test_csv, m3_csv_unificado)

print("\n✅ M3_diagnosis dataset listo para ser dividido en train/val/test")
print(f"📁 Ubicación: {m3_base}")

=== Unificando M3_diagnosis ===

Limpiando carpeta de imágenes existente...
1. Unificando carpetas de imágenes...
Unificación de carpetas completada:
✔️ Imágenes copiadas: 3578
⚠️ Duplicados por nombre ignorados: 0

2. Combinando archivos CSV...
Combinación de CSVs completada:
✔️ Encabezado: ['id_code', 'diagnosis']
✔️ Filas del primer CSV: 2929
✔️ Filas del segundo CSV: 733
✔️ Total de filas en archivo combinado: 3662

✅ M3_diagnosis dataset listo para ser dividido en train/val/test
📁 Ubicación: E:\Proyects Python based\ProyectoAvanzado2\data\processed\M3_diagnosis


In [37]:
print("El todal de imagenes en imagenes es: ", how_many_files(r"E:\Proyects Python based\ProyectoAvanzado2\data\processed\M3_diagnosis\imagenes"))
print("El número de filas del CSV unificado es: ", csv_shape(r"E:\Proyects Python based\ProyectoAvanzado2\data\processed\M3_diagnosis\data_unificado.csv"))

El todal de imagenes en imagenes es:  3578
El número de filas del CSV unificado es:  (3663, 2)
